sql/snapshot.sql

```diff
@@ -0,0 +1,251 @@
+	updated_at timestamp DEFAULT now() NOT NULL,
+	CONSTRAINT metagraph_snapshot_pk PRIMARY KEY (metagraph_id, hash)
+);
+
+
+-- public.dag_balance_change definition
+
+-- Drop table
+
+-- DROP TABLE dag_balance_change;
+
+CREATE TABLE dag_balance_change (
+	snapshot_hash varchar NOT NULL,
+	address varchar NOT NULL,
+	balance int8 NULL,
+	created_at timestamp DEFAULT now() NOT NULL,
+	updated_at timestamp DEFAULT now() NOT NULL,
+	CONSTRAINT dag_balance_change_pk PRIMARY KEY (snapshot_hash, address),
+	CONSTRAINT dag_balance_change_global_snapshot_fk FOREIGN KEY (snapshot_hash) REFERENCES global_snapshot(hash)
+);
+
+
+-- public.dag_block definition
+
+-- Drop table
+
+-- DROP TABLE dag_block;
+
+CREATE TABLE dag_block (
+	hash varchar NOT NULL,
+	snapshot_hash varchar NOT NULL,
+	created_at timestamp DEFAULT now() NOT NULL,
+	updated_at timestamp DEFAULT now() NOT NULL,
+	CONSTRAINT dag_block_pk PRIMARY KEY (hash),
+	CONSTRAINT dag_block_block_fk FOREIGN KEY (hash) REFERENCES block(hash),
+	CONSTRAINT dag_block_global_snapshot_fk FOREIGN KEY (hash) REFERENCES global_snapshot(hash)
+);
+
+
+-- public.dag_reward_transaction definition
+
+-- Drop table
+
+-- DROP TABLE dag_reward_transaction;
+
+CREATE TABLE dag_reward_transaction (
+	destination_addr varchar NOT NULL,
+	amount int8 NOT NULL,
+	global_snapshot_hash varchar NOT NULL,
+	created_at timestamp DEFAULT now() NOT NULL,
+	updated_at timestamp DEFAULT now() NOT NULL,
+	CONSTRAINT reward_transaction_global_snapshot_fk FOREIGN KEY (global_snapshot_hash) REFERENCES global_snapshot(hash)
+);
+
+
+-- public.dag_transaction definition
+
+-- Drop table
+
+-- DROP TABLE dag_transaction;
+
+CREATE TABLE dag_transaction (
+	hash varchar NOT NULL,
+	block_hash varchar NOT NULL,
+	created_at timestamp DEFAULT now() NOT NULL,
+	updated_at timestamp DEFAULT now() NOT NULL,
+	CONSTRAINT dag_transaction_pk PRIMARY KEY (hash),
+	CONSTRAINT dag_transaction_abstract_transaction_fk FOREIGN KEY (hash) REFERENCES abstract_transaction(hash),
+	CONSTRAINT dag_transaction_dag_block_fk FOREIGN KEY (block_hash) REFERENCES dag_block(hash)
+);
+
+
+-- public.fee_transaction definition
+
+-- Drop table
+
+-- DROP TABLE fee_transaction;
+
+CREATE TABLE fee_transaction (
+	hash varchar NOT NULL,
+	amount int8 NOT NULL,
+	source_addr varchar NOT NULL,
+	destination_addr varchar NOT NULL,
+	parent jsonb NOT NULL,
+	metagraph_snapshot_hash varchar NULL,
+	metagraph_id varchar NULL,
+	created_at timestamp DEFAULT now() NOT NULL,
+	updated_at timestamp DEFAULT now() NOT NULL,
+	CONSTRAINT fee_transaction_pkey PRIMARY KEY (hash),
+	CONSTRAINT fee_transaction_metagraph_snapshot_fk FOREIGN KEY (metagraph_id,metagraph_snapshot_hash) REFERENCES metagraph_snapshot(metagraph_id,hash)
+);
+
+
+-- public.metagraph_balance_change definition
+
+-- Drop table
+
+-- DROP TABLE metagraph_balance_change;
+
+CREATE TABLE metagraph_balance_change (
+	metagraph_id varchar NOT NULL,
+	metagraph_hash varchar NOT NULL,
```

Comment:
Same issue as the `dag_balance_change` table. We need ordinal here to be queried, unless the intent is to join to the metagraph_snapshots table. That seems like it might be too inefficient though to search against.